include/flang/Common/idioms.h

```diff
@@ -23,6 +23,9 @@
 #error g++ >= 7.2 is required
 #endif
 
+#include "llvm/Support/ErrorHandling.h"
+#include "llvm/Support/Format.h"
+#include "llvm/Support/raw_ostream.h"
```

Comment:
This file is included in almost every source file in f18. Have you verified that adding these three includes (which include more files) don't have an impact on build speed. Can you get them out of this header?

Reply:
llvm/Support/ErrorHandling.h is needed for llvm_unreachable macro. 
idioms.h files is included in 55+ files and among this 50+ files invoke llvm_unreachable function.
I think this is the best place for this llvm header.
I have removed newer implementation of die function, so rest of the two llvm headers are not needed now.